integration_tests/src/main/python/conftest.py

```diff
@@ -307,6 +301,37 @@ def pytest_configure(config):
 print(f"Starting with OOM injection seed: {oom_random_injection_seed}. "
       "Set env variable SPARK_RAPIDS_TEST_INJECT_OOM_SEED to override.")
 
+# Returns a tuple (seed, permanent) with the seed that test `item` should use given a 
+# possibly defined `datagen_overrides`, and if the seed choice is due to an override, 
+# whether that override is marked as `permanent`
+def get_effective_seed(item, datagen_overrides):
+    if datagen_overrides:
+        is_permanent = False 
+        # if the override is marked as permanent it will always override its seed
+        # else, if the user provides a seed via DATAGEN_SEED, we will override.
+        try:
+            is_permanent = datagen_overrides.kwargs["permanent"]
+        except KeyError:
+            pass
```

Comment:
nit: could use same syntax as L317
```suggestion
        # if the override is marked as permanent it will always override its seed
        # else, if the user provides a seed via DATAGEN_SEED, we will override.
        is_permanent = datagen_overrides.kwargs.get("permanent", False)
```